src/main/business/applicationFacade.js

```diff
@@ -338,14 +261,35 @@ class ApplicationFacade{
 
     /**
      * add meta-question, look for values in MetaQuestion.js
-     *7
      */
-    addMetaQuestion(createMetaQuestionProperties) {
-        return this.metaQuestionController.createMetaQuestion(createMetaQuestionProperties)
+    async addMetaQuestion(data) {
+        validateParameters(data, {
+            keywords: [PRIMITIVE_TYPES.STRING],
+            keys: [{text: PRIMITIVE_TYPES.STRING}],
+            distractors: [{text: PRIMITIVE_TYPES.STRING}],
+            stem: PRIMITIVE_TYPES.STRING,
+        });
+        // case where an appendix is being created alongside the MQ
+        if (data.appendix) {
+            const businessAppendix = await this.metaQuestionController.createAppendix( {
+                ...data.appendix,
```

Comment:
Does FE, send the appendix as an object?

Reply:
- if you use the screen that creates a simple meta-question then no
- if you use the screen that creates a meta-question with an already existing appendix FE will send an extra `appendixTag` key
- if you use the screen that creates an appendix with a meta question FE will send the appendix object

I don't like it that way, and I also want a separate screen to create an appendix only. But this PR is about connecting DAL to the logic that already exists, and not about changing logic.

Reply:
Well base on your answer I guess you did checked that it is integrated without error with the FE, so great...

Reply:
But if not please check it

Reply:
Checked all use-cases. 
I asked Mor/Idan to double check as well.